providers/reddit/reddit.go

```diff
@@ -16,5 +17,13 @@ type Provider struct{}
 
 // BuildURI generates a search URL for Reddit.
 func (p *Provider) BuildURI(q string) string {
+	// If q starts with "r/subreddit query", treat as subreddit and do restrict
+	// search on the subreddit.
+	if len(q) > 3 && q[:3] == "/r/" {
+		subreddit := strings.Split(q, " ")[0]
+		return fmt.Sprintf("https://www.reddit.com%s/search?q=%s"+
+			"&restrict_sr=on&sort=relevance&t=all",
```

Comment:
I think the search will work fine without all these options passed. I would just remove them.